flytekit/clis/sdk_in_container/run.py

```diff
@@ -763,6 +810,87 @@ def get_command(self, ctx, name):
         )
 
 
+class YamlFileReadingCommand(click.RichCommand):
+    def __init__(
+        self,
+        name: str,
+        params: typing.List[click.Option],
+        help: str,
+        callback: typing.Callable = None,
+    ):
+        params.append(
+            click.Option(
+                ["--inputs-file"],
+                required=False,
+                type=click.Path(exists=True, dir_okay=False, resolve_path=True),
+                help="Path to a YAML | JSON file containing inputs for the workflow.",
+            )
+        )
+        super().__init__(name=name, params=params, callback=callback, help=help)
+
+    def parse_args(self, ctx: Context, args: t.List[str]) -> t.List[str]:
+        if "--inputs-file" in args:
+            idx = args.index("--inputs-file")
+            args.pop(idx)
+            f = args.pop(idx)
+            with open(f, "r") as f:
+                try:
+                    inputs = yaml.safe_load(f)
+                except yaml.YAMLError as e:
+                    yaml_e = e
+                    try:
+                        inputs = json.load(f)
+                    except json.JSONDecodeError as e:
+                        raise click.BadParameter(
+                            message=f"Could not load the inputs file. Please make sure it is a valid JSON or YAML file."
+                            f"\n json error: {e},"
+                            f"\n yaml error: {yaml_e}",
+                            param_hint="--inputs-file",
+                        )
+
+            new_args = []
+            for k, v in inputs.items():
+                if isinstance(v, str):
+                    new_args.extend([f"--{k}", v])
+                elif isinstance(v, bool):
+                    if v:
+                        new_args.append(f"--{k}")
+                else:
+                    v = json.dumps(v)
+                    new_args.extend([f"--{k}", v])
+            new_args.extend(args)
+            args = new_args
+        elif not sys.stdin.isatty():
+            f = sys.stdin.read()
+            if f != "":
+                try:
+                    inputs = yaml.safe_load(f)
+                except yaml.YAMLError as e:
+                    yaml_e = e
+                    try:
+                        inputs = json.loads(f)
+                    except json.JSONDecodeError as e:
+                        raise click.BadParameter(
+                            message=f"Could not load the inputs file. Please make sure it is a valid JSON or YAML file."
+                            f"\n json error: {e},"
+                            f"\n yaml error: {yaml_e}",
+                            param_hint="--inputs-file",
+                        )
+                new_args = []
+                for k, v in inputs.items():
```

Comment:
can we refactor this as this is repeated with lines 855-860. You can refactor it to separate block after the yaml has been loaded?